File: backend/app/schemas/rating.py
"""Request/response models for POST /ratings."""

from __future__ import annotations

from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class RatingCreate(BaseModel):
    model_config = ConfigDict(extra="ignore")

    restaurant_id: UUID
    score: int = Field(ge=1, le=5)
    comment: str | None = Field(default=None, max_length=1000)


class RestaurantRatingUpdated(BaseModel):
    id: UUID
    app_rating: float | None
    rating_count: int


class RatingResponse(BaseModel):
    id: UUID
    restaurant_updated: RestaurantRatingUpdated
    points_awarded: int

File: android/app/src/main/java/com/brokenlunch/gr/data/model/Rating.kt
package com.brokenlunch.gr.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class RatingRequest(
    @Json(name = "restaurant_id") val restaurantId: String,
    @Json(name = "score") val score: Int,
    @Json(name = "comment") val comment: String? = null,
)

@JsonClass(generateAdapter = true)
data class RatingResponse(
    @Json(name = "id") val id: String,
    @Json(name = "restaurant_updated") val restaurantUpdated: RestaurantRatingUpdate,
    @Json(name = "points_awarded") val pointsAwarded: Int,
)

@JsonClass(generateAdapter = true)
data class RestaurantRatingUpdate(
    @Json(name = "id") val id: String,
    @Json(name = "app_rating") val appRating: Double,
    @Json(name = "rating_count") val ratingCount: Int,
)
